homeworks/szabolcsgesztesi/2_vars_datatypes/user_info.py

```diff
@@ -0,0 +1,51 @@
+# Táblázat amivel indulunk
+
+user_info = {
+    "name": "Mike",
+    "age": 25,
+    "favourite_meals": ["pizza","carbonara","sushi"],
+    "phone_contacts": {
+        "Mary": "+36701234567",
+        "Tim": "+36207654321",
+        "Tim2": "+36304567321",
+        "Jim": "+364005000"
+    }
+}
+
+# 1. feladat: 
+proglang = input("Adj meg 4 programozási nyelvet vesszővel elválasztva!"  )
+proglang = proglang.split(",")
+user_info["skills"] = proglang
+
+# 2. feladat: 
+user_info["favourite_meals"].sort()
+
+# 3. feladat: 
+print(user_info["favourite_meals"][2])
+
+# 4. feladat: 
+user_info["favourite_meals"].append("spaghetti")
+
+# 5. feladat: itt próbáltam ezt egyetlen sorban megadni de nem ment :) Meg lehetett volna?
+user_info["favourite_meals"].append(user_info["favourite_meals"][2])
+user_info["favourite_meals"].append(user_info["favourite_meals"][3])
+
+# 6. feladat: (sajnos a set randomizálja, az abc-sorrend elveszik)
+user_info["favourite_meals"] = list(set(user_info["favourite_meals"]))
+
+# 7. feladat:
+first = user_info["favourite_meals"][0]
```

Comment:
a logikád jó, viszont ha ezt külön külön csinálod, akkor ugyanaz lesz az első és utolsó elem is, mert előbb felülírod az elsőt, majd ezt a felülírt értéket adod az utolsónak.

user_info["favourite_meals"][0],user_info["favourite_meals"][-1] = user_info["favourite_meals"][-1],user_info["favourite_meals"][0]

próbáld meg így.

Reply:
Igazából miután a saját verziómat elkészítettem a kódból, rákerestem, hogyan lehetett volna mégis egy sorban megírni, és ugyanazt találtam amit írtál, ezt az úgynevezett "tuple cserét", viszont ezt a féle cserét így kifejezetten nem tanultuk, ha jól emlékszem, így arra gondoltam erre nem fog rákérdezni a feladat. Szóval bent hagytam a sajátomat, ami akkor úgy tűnik nem a legjobb megoldás.

Akkor itt meg is kérdezném: A házifeladatok mindig is fognak szándékosan olyat kérdezni amit egy az egyben nem tanultunk, hanem kifejezetten saját utánajárás szükséges hozzá, igaz? Ha így van az szuper, mert ettől még izgalmasabb lesz a dolog! Csak eddig úgy gondoltam ez nem igazán elvárás egyelőre, de ha így van az jó! :)